src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -762,7 +761,67 @@ Copyright (c) .NET Foundation. All rights reserved.
       </_NoneWithTargetPath>
 
     </ItemGroup>
+  </Target>
+  
+  <!-- Determine the managed assembly subset of ResolvedFileToPublish that should be post-processed by linker, and ready to run compilation -->
+  <!-- -->
+  <Target Name="_ComputeAssembliesToPostprocessOnPublish" 
+          DependsOnTargets="_ComputeUserRuntimeAssemblies">
+    
+    <!-- 
+      Default set of files to post-process correspond to the items that would be designated
+      as managed runtime assemblies in .deps.json, and published to root of the application.
+      RuntimeTargets and satellite assemblies are excluded. Currently, both linker and ready
+      to run require a RID, so there will not be RuntimeTargets. Linker could conceptually 
+      operate without a RID, but would not know how to handle multiple assemblies with same
+      identity.
+    -->
+    <ItemGroup>
+      <!-- Assemblies from packages -->
+      <_ManagedRuntimeAssembly Include="@(RuntimeCopyLocalItems)" />
+      
+      <!-- Assemblies from other references -->
+      <_ManagedRuntimeAssembly Include="@(UserRuntimeAssembly)" />
+      
+      <!-- Assembly produced by this project -->
+      <_ManagedRuntimeAssembly Include="@(IntermediateAssembly)" />
+    </ItemGroup>
+    
+    <!-- Assemblies from runtime packs for self-contained apps -->
+    <CallTarget Targets="_ComputeManagedRuntimePackAssemblies" Condition="'$(SelfContained)' == 'true'">
+      <Output TaskParameter="TargetOutputs" ItemName="_ManagedRuntimeAssembly" />
```

Comment:
Did you get an answer on whether or not this was the recommended way to conditionally depend on a target?

I think you can also conditionally set a property that you use for DependsOnTargets.

Reply:
I’ll do that